### ISSUE TITLE
RAG的问答能力很差。

### ISSUE THREAD

**Issue by HouYueJie** [FlagAI-Open/Aquila2#107]
RAG效果很差，是不是提问方式不对？有做过文档问答的测试么？
![image](https://github.com/FlagAI-Open/Aquila2/assets/66465115/8311d403-fa0d-40d0-92a8-d9840f0bc77d)

**Comment by HouYueJie** [comment]
补充一下，用的模型是“AquilaChat2-34B v1.2 ”

**Comment by ftgreat** [comment]
麻烦贴下文本，我们看下这个case

**Comment by HouYueJie** [comment]
from transformers import AutoTokenizer, AutoModelForCausalLM
from transformers import BitsAndBytesConfig
import torch

device = torch.device("cuda")
model_info = "./"
tokenizer = AutoTokenizer.from_pretrained(model_info, trust_remote_code=True)
model = AutoModelForCausalLM.from_pretrained(model_info,device_map='auto', trust_remote_code=True, torch_dtype=torch.bfloat16,)
model.eval()
text = """结合已知要点，回答问题。
已知要点:免疫组化类体外诊断试剂是指用于检测冰冻切片、石蜡包埋组织切片及细胞制片等标本的抗体试剂及相关产品。该类产品根据其预期用途可以分为不同的管理类别如用于肿瘤诊断的按照第三类医器械管理
，用于自身免疫病的按照第二类医疗器械管理。免疫组化检测用样本处理试剂按照第一类医疗器械管理。双抗体多抗体联合检测的试剂按照组合类产品管理。根据临床专业指南或专家共识，可以用于诊断或治疗的>免疫组化类体外诊断试剂被认为具有明确诊断价值。
问题:免疫组化是几类医疗器械?
"""
from predict import predict
out = predict(model, text, tokenizer=tokenizer, max_gen_len=256, top_p=0.9,
              seed=123, topk=15, temperature=1.0, sft=True, device=device,
              model_name="AquilaChat2-34B")
print(out)

**Comment by zacliu2023** [comment]
16K版本感觉有时候是可以出来的。
现在版本还没有针对这种类型的任务进行优化过，后续可以针对这种类型的任务进行优化。

<img width="909" alt="image" src="https://github.com/FlagAI-Open/Aquila2/assets/140705676/ab7b23c2-e525-4323-89fb-ac836d175789">

**Comment by HouYueJie** [comment]
是的，有时候可以。期待优化后的版本。
